docs/notebooks/deep_gaussian_processes.pct.py

```diff
@@ -0,0 +1,354 @@
+# %% [markdown]
+# # Using deep Gaussian processes with GPflux for Bayesian optimization.
+
+# %%
+import numpy as np
+import tensorflow as tf
+
+np.random.seed(1794)
+tf.random.set_seed(1794)
+tf.keras.backend.set_floatx("float64")
+
+# %% [markdown]
+# ## Describe the problem
+# In this example, we look to find the minimum value of the two-dimensional Branin function over the hypercube $[0, 1]^2$, modified with a discontinuity. We can represent the search space using a `Box`, and plot contours of the Branin over this space.
```

Comment:
branin? perhaps explain why particular function, why is that an appropriate example for DGPs?